crypto/bls_test.go

```diff
@@ -51,6 +51,34 @@ func TestBLSMainMethods(t *testing.T) {
 		require.NoError(t, err)
 		assert.False(t, valid)
 	})
+
+	t.Run("private key equal to 1 and -1", func(t *testing.T) {
+		sk1Bytes := make([]byte, PrKeyLenBLSBLS12381)
+		sk1Bytes[PrKeyLenBLSBLS12381-1] = 1
+		sk1, err := DecodePrivateKey(BLSBLS12381, sk1Bytes)
+		require.NoError(t, err)
+
+		skMinus1Bytes := make([]byte, PrKeyLenBLSBLS12381)
+		copy(skMinus1Bytes, BLS12381Order)
+		skMinus1Bytes[PrKeyLenBLSBLS12381-1] -= 1
+		skMinus1, err := DecodePrivateKey(BLSBLS12381, skMinus1Bytes)
+		require.NoError(t, err)
+
+		for _, sk := range []PrivateKey{sk1, skMinus1} {
+			input := make([]byte, 100)
+			_, err = mrand.Read(input)
+			require.NoError(t, err)
+			s, err := sk.Sign(input, hasher)
+			require.NoError(t, err)
+			pk := sk.PublicKey()
+
+			// test a valid signature
+			result, err := pk.Verify(s, input, hasher)
+			assert.NoError(t, err)
```

Comment:
```suggestion
			require.NoError(t, err)
```

Reply:
Well, the purpose of the test is to make sure the function does not error in this case. Before this PR's fix, the function errors in that case while it shouldn't.

Reply:
Whether to use `assert` or `require` depends on whether it is useful/meaningful to also check subsequent assertions, assuming this one failed. In this case it probably isn't useful (typically an error return means that value return `result` isn't meaningful). In other words, this test output:
```
Expected no error ...
Expected result to be true
```

is not more useful than just
```
Expected no error ...
```

In this case it doesn't make much difference IMO because there is only one subsequent assertion.